test/suites/network_ovn.sh

```diff
@@ -0,0 +1,225 @@
+  assert_row_count() {
+    for table_name in "${!tables[@]}"; do
+      count="${tables[${table_name}]}"
+      echo "Checking ${table_name} has ${count} rows..."
+      [ "$(ovn-nbctl --format csv --no-headings list "${table_name}" | wc -l)" = "${count}" ]
+    done
+  }
+
+  # Validate northbound database is initially empty (so we don't inadvertently break anything).
+  reset_row_count
+  assert_row_count
+
+  lxc config set network.ovn.northbound_connection "${LXD_OVN_NB_CONNECTION}"
+
+  # If the connection uses SSL we have more required environment variables.
+  # Set the client cert, key, and CA cert.
+  if [[ "${LXD_OVN_NB_CONNECTION}" =~ ^ssl: ]]; then
+    lxc config set network.ovn.client_cert="$(< "${LXD_OVN_NB_CLIENT_CRT_FILE}")"
+    lxc config set network.ovn.client_key="$(< "${LXD_OVN_NB_CLIENT_KEY_FILE}")"
+    lxc config set network.ovn.ca_cert="$(< "${LXD_OVN_NB_CA_CRT_FILE}")"
+  fi
+
+  # Create a bridge for use as an uplink.
+  uplink_network="uplink$$"
+  lxc network create "${uplink_network}" \
+      ipv4.address=10.10.10.1/24 ipv4.nat=true \
+      ipv4.dhcp.ranges=10.10.10.2-10.10.10.199 \
+      ipv4.ovn.ranges=10.10.10.200-10.10.10.254 \
+      ipv6.address=fd42:4242:4242:1010::1/64 ipv6.nat=true \
+      ipv6.ovn.ranges=fd42:4242:4242:1010::200-fd42:4242:4242:1010::254 \
+      ipv4.routes=192.0.2.0/24 ipv6.routes=2001:db8:1:2::/64
+
+  # Create an OVN network.
+  ovn_network="ovn$$"
+  lxc network create "${ovn_network}" --type ovn network="${uplink_network}" \
+      ipv4.address=10.24.140.1/24 ipv4.nat=true \
+      ipv6.address=fd42:bd85:5f89:5293::1/64 ipv6.nat=true
+
+  # Check this created the correct number of entries.
+  tables["ACL"]=15
+  tables["Address_Set"]=2
+  tables["DHCP_Options"]=2
+  tables["HA_Chassis"]=1
+  tables["HA_Chassis_Group"]=1
+  tables["Logical_Router"]=1
+  tables["Logical_Router_Policy"]=3
+  tables["Logical_Router_Port"]=2
+  tables["Logical_Router_Static_Route"]=2
+  tables["Logical_Switch"]=2
+  tables["Logical_Switch_Port"]=3
+  tables["NAT"]=2
+  tables["Port_Group"]=1
+  assert_row_count
+
+  ovn_network_id="$(lxd sql global --format csv "SELECT id FROM networks WHERE name = '${ovn_network}'")"
+
+  # Check expected chassis and chassis group are created.
+  chassis_group_name="lxd-net${ovn_network_id}"
+  chassis_id="$(sudo ovn-nbctl --format json get ha_chassis_group "${chassis_group_name}" ha_chassis | tr -d '[]')"
+  sudo ovn-nbctl get ha_chassis "${chassis_id}" priority
+
+  # Check expected logical router has the correct name.
+  logical_router_name="${chassis_group_name}-lr"
+  ovn-nbctl get logical_router "${logical_router_name}" options
+
+  # Check external logical router port exists and has default gateway MTU.
+  external_router_port_name="${logical_router_name}-lrp-ext"
+  [ "$(ovn-nbctl get logical_router_port "${external_router_port_name}" options:gateway_mtu)" = '"1500"' ]
```

Comment:
In CI, this fails as the effective MTU is 1442 which sounds like reasonable overhead unless we use baby jumbo packets for the transport layer?